src/arch.rs

```diff
@@ -15,7 +15,7 @@
 //! > Having community-created `Arch` implementations distributed in a separate
 //! crate helps minimize any unnecessary "version churn" in `gdbstub` core.
 
-use core::fmt::Debug;
+use core::{fmt::Debug, num::NonZeroUsize};
```

Comment:
nit: imports should only be merged at the "leaf" level:

```suggestion
use core::fmt::Debug;
use core::num::NonZeroUsize;
```